packit_service/models.py

```diff
@@ -463,11 +486,16 @@ def get_by_build_id(
             #   You might need to add explicit type casts.
             build_id = str(build_id)
         with get_sa_session() as session:
-            return (
-                session.query(CoprBuildModel)
-                .filter_by(build_id=build_id, target=target)
-                .first()
-            )
+            if target:
+                return (
+                    session.query(CoprBuildModel)
+                    .filter_by(build_id=build_id, target=target)
+                    .first()
+                )
+            else:
+                return (
+                    session.query(CoprBuildModel).filter_by(build_id=build_id).first()
+                )
```

Comment:
You can use the `Query` object returned by `session.query()` and do multiple rounds of filtering. If my understanding is correct, SQLAlchemy does the query only once, when the scope of the session is about to end.

```python
        with get_sa_session() as session:
            query = session.query(CoprBuildModel).filter_by(build_id=build_id)
            if target:
                query = query.filter_by(target=target)
            return query.first()
```

Also, I would rather set the default of `target` to `None` to allow calling this without a target.

Reply:
> You can use the Query object returned by session.query() and do multiple rounds of filtering

Thanks

> Also, I would rather set the default of target to None to allow calling this without a target.

Will do